cpp/src/strings/convert/convert_urls.cu

```diff
@@ -222,30 +276,65 @@ std::unique_ptr<column> url_decode(
   size_type strings_count = strings.size();
   if (strings_count == 0) return make_empty_strings_column(stream, mr);
 
-  auto strings_column = column_device_view::create(strings.parent(), stream);
-  auto d_strings      = *strings_column;
+  auto d_offsets = strings.offsets().data<size_type>();
+  // use indices relative to the base column data so they can be compared to offset values
+  auto d_in_chars  = strings.chars().head<char>();
+  auto chars_bytes = strings.chars_size();
+  // determine index of first character in base column
+  size_type chars_start = 0;
+  if (strings.offset() != 0) {
+    chars_start = cudf::detail::get_value<size_type>(strings.offsets(), 0, stream);
+  }
+  size_type chars_end = chars_start + chars_bytes;
```

Comment:
This is not accounting for sliced columns correctly. 
The `d_in_chars` and `d_offsets` are pointing to the beginning of the child columns. Also, `chars_bytes` is the size of the child column ignoring the parent's offset.
We probably should add some gtests with sliced columns -- offset and size.

Reply:
Yep, I definitely need to add sliced string tests.

Are the docs for `chars_size()` wrong?
```c++
  /**
   * @brief Returns the number of bytes in the chars child column.
   *
   * This accounts for the offset of the strings' column_view and
   * for empty columns.
   */
  size_type chars_size() const noexcept;
```

Reply:
Yes. That doc comment is wrong.